base/core/include/motis/core/common/logging.h

```diff
@@ -106,4 +106,17 @@ struct manual_timer final {
   std::chrono::time_point<std::chrono::steady_clock> start_;
 };
 
+inline void clog_set_progress_bounds(double output_low = 0.,
+                                     double output_high = 100.,
+                                     double input_high = 100.) {
+  std::clog << '\0' << 'B' << output_low << " " << output_high << " "
+            << input_high << '\0';
+}
+
+inline void clog_update_progress_int(size_t progress, size_t rate_limit = 1) {
```

Comment:
Warum "int"? Wir haben doch sonst auch keine Typen im Funktionsnamen?

Wir das `rate_limit` wirklich irgendwo benötigt?

```suggestion
inline void import_update_progress(size_t progress, size_t rate_limit = 1) {
```

Reply:
int weil der rest double ist ... aber kann man auch wegmachen

und naja die funktion wird ein paar mal millionen mal aufgerufen
womit wir bei "kleinvieh macht auch mist" wären ...

Reply:
Dann sollte man vll. die Anzahl der Calls reduzieren?
Wie hier: https://github.com/motis-project/motis/blob/master/base/loader/src/hrd/hrd_parser.cc#L162

Reply:
macht das doch?

Reply:
wäre natürlich schöner, wenn man am ende nur calls an clog macht, wenn sich ne ganze prozentzahl an der gesamtausgabe getan hat. aber das geht wahrscheinlich nicht so einfach?

weil so muss man ja wissen, wie viele elemente "typisch" sind und das hängt wiederum vom datensatz ab

Reply:
naja wie viel ein prozent ist weiß nur das andere ende ... außer man baut da noch einen wrapper, aber ich glaube das wäre overkill

Für diese variante reicht es, wenn man weiß ob ein element "schnell" oder "langsam" ist.
Wenn ein element schnell ist nimmt man ne größere zahl, wenn ein element langsam ist nimmt man eine kleinere zahl.

So wie es da steht ist es glaube ich ein ganz guter tradeoff zwischen simpler implementierung und absolutem performance impact.

Aber wenn man sich mit ein bisschen bauchgefühl die calls um nen faktor von 1k-10k reduzieren kann kommt man vielleicht schonmal nicht ganz so schnell in eine so peinliche situation wie npm damals.

Reply:
so wie bei npm sollte es nicht werden

```cpp
  if (update(name, s)) {
    print();
  }
```

solange alle print calles per if(..has changed..) geschützt sind, wird nicht unnötige I/O gemacht